Directory.Build.props

```diff
@@ -212,4 +212,8 @@
       <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
     </PackageReference>
   </ItemGroup>
+
+  <PropertyGroup Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net8.0'))">
```

Comment:
nit: maybe add a comment that we expect net8, net9 and future net to be AOTCompatible, but not netstandard or netfwk

the IsTargetFrameworkCompatible is quite new and non-obvious https://learn.microsoft.com/en-us/visualstudio/msbuild/property-functions?view=vs-2022#TargetFramework

Reply:
IsTargetFrameworkCompatible is the recommended way to handle this as AOT compatibility was added in .NET 7 IIRC.